GeofenceDisplay/src/geofence-display.ts

```diff
@@ -2,153 +2,82 @@ export {}
-    //   let geofence = new Geofence(ent, ctx)
-    //   self.geofence.map[ent.id] =
-    //     self.config.geofence.prepare(geofence) || geofence
-    // })
-
-    // // GeofencePane related code
-    // // TODO: Tie in with existing setup
-    // _map.createPane('geofence')
-    // let geofencePane = _map.getPane('geofence')
-    // geofencePane.style.zIndex = 230
-    // geofencePane.style.pointerEvents = 'none'
-
-    // _map.createPane('geofenceLabel')
-    // let geofenceLabelPane = _map.getPane('geofenceLabel')
-    // geofenceLabelPane.style.zIndex = 235
-    // geofenceLabelPane.style.pointerEvents = 'none'
-
-    // self.layer.geofence = L.layerGroup(null, { pane: 'geofence' }).addTo(_map)
-    // self.layer.geofence.name$ = 'geofence'
-
-    // //
-
-    // // TODO: show/clear geofence functionality required here?
-    // self.showGeofence = function (geofence, show) {
-    //   if (null == geofence) {
-    //     return
-    //   }
-
-    //   show = !!show
-
-    //   if (true === show) {
-    //     geofence.show(self.layer.geofence)
-    //   } else if (false === show) {
-    //     geofence.hide()
-    //   }
-    // }
-
-    // self.clearGeofences = function () {
-    //   for (let geofenceID in self.geofence.map) {
-    //     let geofence = self.geofence.map[geofenceID]
-    //     delete self.geofence.map[geofenceID]
-    //     if (geofence && geofence.hide) {
-    //       geofence.hide()
-    //     }
-    //   }
-    // }
-
-    // // TODO: add geofencePane to div
-
-    return _map
+    console.log('GeofenceDisplay onAdd function called.')
+    let self = this
+    console.log('Local geofence variable before add:', self._state.geofenceByID)
+    for (let geodata in self.options.geofences) {
+      let geo = self.options.geofences[geodata]
+      let geofence = new Geofence(geo, null)
+      self._state.geofenceByID[geo.id] = geofence
+      geofence.show(_map)
+    }
+    console.log('Local geofence variable after add:', self._state.geofenceByID)
   },
 
   onRemove: function (this: any, _map: any) {
-    console.log('PGeoD onRemove')
-  },
-
-  clear: function (this: any) {
-    console.log('PGeoD clear')
+    console.log('GeofenceDisplay onRemove function called.')
+    let self = this
+    console.log(
+      'Local geofence variable before remove:',
+      self._state.geofenceByID
+    )
+    for (let id in self._state.geofenceByID) {
+      self._state.geofenceByID[id].hide()
+      delete self._state.geofenceByID[id]
+    }
+    console.log(
+      'Local geofence variable after remove:',
+      self._state.geofenceByID
+    )
   },
 })
 
 class Geofence {
-  ent = null
-  ctx = null
-  poly = L.polygon(
-    [
-      [37, -109.05],
-      [41, -109.03],
-      [41, -102.05],
-      [37, -102.04],
-    ],
-    { color: 'red' }
-  )
+  ent: any = null
+  ctx: any = null
+  poly: any = null
 
   constructor(ent: any, ctx: any) {
     this.ent = ent
     this.ctx = ctx
   }
 
-  show(layer: any) {
+  show(map: any) {
```

Comment:
refactor so that `show` does not need any arguments
thus `map` should come in via the `ctx` context argument

Reply:
Done 👍